plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -29,31 +32,103 @@ import (
 )
 
 type strategyStore struct {
+	// to allow concurrent update of sampling strategies
+	lock sync.RWMutex
+
 	logger *zap.Logger
 
 	defaultStrategy   *sampling.SamplingStrategyResponse
 	serviceStrategies map[string]*sampling.SamplingStrategyResponse
 }
 
+func (h *strategyStore) loadAndParseStrategies(strategiesFile string) error {
+	s, err := loadStrategies(strategiesFile)
+	if err != nil {
+		h.logger.Warn("Using the last saved configuration for sampling strategies.", zap.Error(err))
+		return err
+	}
+
+	h.logger.Info("Updating sampling strategies file!", zap.Any("Strategies", s))
+	h.parseStrategies(s)
+	return nil
+}
+
 // NewStrategyStore creates a strategy store that holds static sampling strategies.
 func NewStrategyStore(options Options, logger *zap.Logger) (ss.StrategyStore, error) {
 	h := &strategyStore{
 		logger:            logger,
 		serviceStrategies: make(map[string]*sampling.SamplingStrategyResponse),
 	}
-	strategies, err := loadStrategies(options.StrategiesFile)
-	if err != nil {
+
+	// Read strategies
+	if err := h.loadAndParseStrategies(options.StrategiesFile); err != nil {
 		return nil, err
 	}
-	h.parseStrategies(strategies)
+
+	// Watch strategies file for changes.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("failed to create a new watcher for the sampling strategies file", zap.Error(err))
+	}
+
+	go h.runWatcherLoop(watcher, options.StrategiesFile)
+
+	if err = watcher.Add(options.StrategiesFile); err != nil {
```

Comment:
```suggestion
	if err := watcher.Add(options.StrategiesFile); err != nil {
```
don't leak variable scope